lib/widgets/path_editor/editors/graph_editor.dart

```diff
@@ -0,0 +1,427 @@
+    );
+
+    return SideTitleWidget(
+        axisSide: meta.axisSide,
+        space: 10,
+        child: Text(value.toInt().toString(),
+            style: style, textAlign: TextAlign.center));
+  }
+
+  Widget bottomTitleWidgets(double value, TitleMeta meta) {
+    if (((value * 10.0) - (value * 10.0).round().toInt()).abs() > 0.01) {
+      return const Text('');
+    }
+
+    const style = TextStyle(
+      color: Colors.white,
+      fontWeight: FontWeight.bold,
+      fontSize: 16,
+    );
+
+    return SideTitleWidget(
+      axisSide: meta.axisSide,
+      space: 10,
+      child: Text(value.toStringAsFixed(1), style: style),
+    );
+  }
+}
+
+class GraphEditor extends StatefulWidget {
+  final RobotPath path;
+  final bool holonomicMode;
+  final void Function(RobotPath path) savePath;
+  final SharedPreferences prefs;
+  static String prefShowVelocity = 'showVelocity';
+  static String prefShowAccel = 'showAccel';
+  static String prefShowHeading = 'showHeading';
+  static String prefShowRotation = 'showRotation';
+  static String prefShowAngularVelocity = 'showAngularVelocity';
+  static String prefShowCurvature = 'showCurvature';
+  static Color colorVelocity = Colors.red;
+  static Color colorAccel = Colors.orange;
+  static Color colorHeading = Colors.yellow;
+  static Color colorRotation = Colors.yellow;
+  static Color colorAngularVelocity = Colors.green;
+  static Color colorCurvature = Colors.blue;
+
+  const GraphEditor(
+      {required this.path,
+      required this.holonomicMode,
+      required this.savePath,
+      required this.prefs,
+      super.key});
+
+  @override
+  State<GraphEditor> createState() => _GraphEditorState();
+}
+
+class _GraphEditorState extends State<GraphEditor> {
+  final GlobalKey _key = GlobalKey();
+  UniqueKey _graphRuntimeKey = UniqueKey();
+  bool isSampled = false;
+  bool cardMinimized = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return Stack(
+      key: _key,
+      children: [
+        Padding(
+          padding: const EdgeInsets.fromLTRB(48, 48, 48, 72),
+          child: _LineChart(
+              isSampled: isSampled,
+              path: widget.path,
+              holonomicMode: widget.holonomicMode,
+              prefs: widget.prefs),
+        ),
+        _buildGraphSettingsCard(),
+        _buildGeneratorSettingsCard(),
+      ],
+    );
+  }
+
+  Widget _buildGraphSettingsCard() {
+    return GraphSettingsCard(
+        stackKey: _key,
+        key: _graphRuntimeKey,
+        holonomicMode: widget.holonomicMode,
+        onUpdate: () async {
+          setState(() {
+            // Force rebuild card to update runtime
+            _graphRuntimeKey = UniqueKey();
+          });
+        },
+        onToggleSampled: () async {
+          setState(() {
+            isSampled = !isSampled;
+          });
+        },
+        onToggleMinimized: () async {
+          setState(() {
+            cardMinimized = !cardMinimized;
+          });
+        },
```

Comment:
Doesn't need to be async since you're not running any async code

```suggestion
        onToggleMinimized: () {
          setState(() {
            _cardMinimized = !_cardMinimized;
          });
        },
```